src/versionedTypes/ts47-mergeParameters.ts

```diff
@@ -1,23 +1,21 @@
 // This entire implementation courtesy of Anders Hjelsberg:
 // https://github.com/microsoft/TypeScript/pull/50831#issuecomment-1253830522
 
+import { ReverseHead, ReverseTail } from '../types'
+
 type UnknownFunction = (...args: any[]) => any
 
-type LongestTuple<T extends readonly unknown[][]> = T extends [
```

Comment:
The `readonly unknown[][] `requirement has to be removed from several places since `ReverseHead` and `ReverseTail` use recursive types to find the first an last tuple element. Typescript complains when a condition (like `readonly unknown[][]`) has to to be compared with a type that is recursive, since it can't check the result of that type.

The checks of whether a type is `readonly unknown[][]` are moved to conditintional types instead where it is needed. 

These are internal types specifically used for this use case, so removing the requirement should not be an issue.